File: solutions/cpp/pangram/3/pangram.cpp
#include "pangram.h"

#include <bitset>

namespace pangram {

    bool is_pangram(const char *str) {
        std::bitset<26> mask;
        for (; *str; str++) {
            if (*str >= 'A' && *str <= 'Z')
                mask.set(*str - 'A');
            if (*str >= 'a' && *str <= 'z')
                mask.set(*str - 'a');
        }
        return mask.all();
    }

}  // namespace pangram

File: solutions/ruby/pangram/2/pangram.rb
#!/usr/bin/env ruby

module BookKeeping
  VERSION = 2
end

#
module Pangram
  LOWER_ALPHABET = 'a'..'z'

  class << self
    def pangram?(sentence)
      downcased_sentence = sentence.downcase
      LOWER_ALPHABET.all? { |c| downcased_sentence.include? c }
    end
    alias is_pangram? pangram?
  end
end